persistence-modules/java-mongodb/src/test/java/com/baeldung/MongodbCrudOperationsLiveTest.java

```diff
@@ -0,0 +1,200 @@
+
+        InsertOneResult result = collection.insertOne(newMovie);
+        assertTrue(result.wasAcknowledged(), "Insert operation should be acknowledged");
+
+        Document found = collection.find(eq("_id", objectId))
+            .first();
+        assertNotNull(found, "Document should be found in the database");
+        assertEquals("Silly Video", found.getString("title"), "Titles should match");
+    }
+
+    @Test
+    public void givenMultipleDocuments_whenInsert_thenShouldBeInserted() {
+        List<Document> movies = Arrays.asList(new Document().append("title", "Silly Video 2"), new Document().append("title", "Silly Video: The Prequel"));
+
+        InsertManyResult result = collection.insertMany(movies);
+        assertTrue(result.wasAcknowledged(), "Insert operation should be acknowledged");
+
+        Document foundMovie1 = collection.find(new Document("title", "Silly Video 2"))
+            .first();
+        Document foundMovie2 = collection.find(new Document("title", "Silly Video: The Prequel"))
+            .first();
+
+        assertNotNull(foundMovie1, "First document should be found in the database");
+        assertNotNull(foundMovie2, "Second document should be found in the database");
+    }
+
+    @Test
+    public void givenTitle_whenFind_thenDocumentShouldBeFound() {
+        Document found = collection.find(eq("title", "The Great Train Robbery"))
+            .first();
+        assertNotNull(found);
+        assertEquals("The Great Train Robbery", found.getString("title"));
+    }
+
+    @Test
+    public void givenProjectionAndSort_whenFind_thenDocumentShouldBeFound() {
+        Bson projection = Projections.fields(Projections.include("title", "genres"), Projections.excludeId());
+        Document found = collection.find(eq("title", "The Great Train Robbery"))
+            .projection(projection)
+            .sort(ascending("year"))
+            .first();
+        assertNotNull(found);
+        assertEquals("The Great Train Robbery", found.getString("title"));
+    }
+
+    @Test
+    public void givenProjectionAndSort_whenFindMany_thenDocumentsShouldBeFound() {
+        Bson projection = Projections.fields(Projections.include("title", "year"), Projections.excludeId());
+        MongoCursor<Document> cursor = collection.find(gt("year", 1900))
+            .projection(projection)
+            .sort(ascending("title"))
+            .iterator();
+        List<String> titles = new ArrayList<>();
+        try {
+            while (cursor.hasNext()) {
+                Document doc = cursor.next();
+                titles.add(doc.getString("title"));
+                assertNotNull(doc.getString("title"));
+                assertNotNull(doc.getInteger("year"));
+                assertNull(doc.get("_id"));
+            }
+        } finally {
+            cursor.close();
+        }
+        if (titles.size() > 1) {
+            for (int i = 0; i < titles.size() - 1; i++) {
+                assertTrue(titles.get(i)
```

Comment:
I just noticed we have quite a few of these complex assertTrue assertions

We have a guideline to switch to using AssertJ if the assertions start to get complex like this - can we refactor the assertions to use AssertJ throughout please?

Reply:
@timotheekelly ping here

Reply:
Yes I can refactor this now, thank you

Reply:
I've refactored to use the AssertJ. I'm somewhat unfamiliar, but I believe these tests follow best practice